File: jsimple-oauth/src/main/java/jsimple/oauth/utils/OAuthEncoder.java
package jsimple.oauth.utils;

import jsimple.net.UrlDecoder;
import jsimple.net.UrlEncoder;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

/**
 * @author: Pablo Fernandez
 */
public class OAuthEncoder {
    public static String encode(String plain) {
        String encoded = UrlEncoder.encode(plain);
        encoded = encoded.replace("*", "%2A");
        encoded = encoded.replace("+", "%20");
        encoded = encoded.replace("%7E", "~");
        return encoded;
    }

    public static String decode(String encoded) {
        return UrlDecoder.decode(encoded);
    }
}

File: jsimple-oauth/c#/jsimple/oauth/utils/OAuthEncoder.cs
namespace jsimple.oauth.utils
{

	using UrlDecoder = jsimple.net.UrlDecoder;
	using UrlEncoder = jsimple.net.UrlEncoder;

	/// <summary>
	/// @author: Pablo Fernandez
	/// </summary>
	public class OAuthEncoder
	{
		public static string encode(string plain)
		{
			string encoded = UrlEncoder.encode(plain);
			encoded = encoded.Replace("*", "%2A");
			encoded = encoded.Replace("+", "%20");
			encoded = encoded.Replace("%7E", "~");
			return encoded;
		}

		public static string decode(string encoded)
		{
			return UrlDecoder.decode(encoded);
		}
	}

}